x/nexus/keeper/msg_server.go

```diff
@@ -84,6 +87,10 @@ func (s msgServer) DeregisterChainMaintainer(c context.Context, req *types.Dereg
 		return nil, fmt.Errorf("account %v is not registered as a validator proxy", req.Sender.String())
 	}
 
+	if s.staking.Validator(ctx, validator) == nil {
```

Comment:
So one cannot deregister after unbonding? Why?

Reply:
Yeah, I was wondering whether to allow that. My thought process was that we really need to have a notion of whether a chain maintainer is active. If a chain maintainer stops voting for a long time, we should just remove them from the set. Otherwise, as noted in the rewards related bug, they eat up the rewards from other validators (even if it's never released to them).
This way we don't have to rely on a validator deregistering even after they're unbonded.
Shall I keep the check for consistency or remove it?

Reply:
"This way we don't have to rely on a validator deregistering even after they're unbonded." I don't understand this. With this check added, one cannot deregister anymore after unbonding which means the only way for him/her to fix it is to re-bond, deregister, and then unbond again. I can totally see that there probably should exist a chain maintainer jail, but that would be a separate feature.

Reply:
I removed it in this place. I do think validators will forget to deregister regardless of whether they're still a validator or not. So, the chain maintainer auto deactivation/jail would be nice to have.